thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/onboard/tasks/FunctionCreationOnboardingTask.java

```diff
@@ -148,13 +147,15 @@ public void run() {
     }
     TimeSpec timeSpec = ThirdEyeUtils.getTimeSpecFromDatasetConfig(datasetConfig);
     TimeGranularity dataGranularity = timeSpec.getDataGranularity();
-    TimeGranularity userAssignedDataGranularity = null;
-    try{
-      userAssignedDataGranularity = TimeGranularity.fromString(configuration.getString(DATA_GRANULARITY));
-    } catch (Exception e) {
-      LOG.warn("Unable to parse user input data granularity: {}; use default from dataset config", configuration.getString(DATA_GRANULARITY));
-    }
-    if (userAssignedDataGranularity != null) {
+    if (configuration.containsKey(DATA_GRANULARITY)) {
+      TimeGranularity userAssignedDataGranularity = null;
+      try {
+        userAssignedDataGranularity = TimeGranularity.fromString(configuration.getString(DATA_GRANULARITY));
+      } catch (Exception e) {
+        LOG.warn("Unable to parse user input data granularity: {}; use default from dataset config",
```

Comment:
The log message is incorrect because we doesn't continue with the default granularity.

Reply:
Thanks for pointing this out.